packages/cursorless-vscode/src/ide/vscode/hats/VscodeHatRenderer.ts

```diff
@@ -257,22 +262,24 @@ export default class VscodeHatRenderer {
     const newViewBoxWidth = originalViewBoxWidth * (svgWidthPx / hatWidthPx);
     const newViewBoxHeight = newViewBoxWidth * (svgHeightPx / svgWidthPx);
     const newViewBoxX =
-      (-(characterWidth - hatWidthPx) * (newViewBoxWidth / svgWidthPx)) / 2;
+      (-(characterWidth - hatWidthPx * widthFactor) *
+        (newViewBoxWidth / svgWidthPx)) /
+      2;
     const newViewBoxY = 0;
 
     const newViewBoxString = `${newViewBoxX} ${newViewBoxY} ${newViewBoxWidth} ${newViewBoxHeight}`;
 
-    if (
-      rawSvg.match(/width="[^"]+"/) == null ||
-      rawSvg.match(/height="[^"]+"/) == null
-    ) {
-      throw Error("Raw svg doesn't have height or width");
-    }
-
-    const svg = rawSvg
-      .replace(/width="[^"]+"/, `width="${svgWidthPx}px"`)
-      .replace(/height="[^"]+"/, `height="${svgHeightPx}px"`)
-      .replace(/viewBox="([^"]+)"/, `viewBox="${newViewBoxString}"`);
+    const innerSvg = rawSvg
+      .replace(/width="[^"]+"/, ``)
+      .replace(/height="[^"]+"/, ``)
+      .replace(/viewBox="([^"]+)"/, `style="overflow:visible"`);
+
+    const svg =
+      `<svg xmlns="http://www.w3.org/2000/svg" ` +
+      `viewBox="${newViewBoxString}" ` +
+      `width="${svgWidthPx}px" ` +
+      `height="${svgHeightPx}px">` +
+      `<g transform="scale(${widthFactor}, 1)">${innerSvg}</g></svg>`;
```

Comment:
Had a problem here: If the `rawSvg` string includes an xml-header like
```
<?xml version="1.0" encoding="UTF-8"?><svg ...
```
Then the hat will not be displayed. I also had problems if the svg has a trailing newline or if it is formatted with indentation, but I could not pin down why that would cause problems.

Reply:
I can make a PR for adding xml-header stripping to the preprocess script if you would like

Reply:
Interesting.  Yeah feel free to add to preprocessor.  We also bundle an xml parser for use in cheatsheet so should prob use that to upgrade to something more robust.  Feel free to take a crack at it or even just file an issue